docs/add-runtime.md

```diff
@@ -0,0 +1,254 @@
+
+The first thing you need to do is find a docker image to base your runtime off, You can find these at [Docker Hub](https://hub.docker.com). If possible try to use verified official builds of the language you are creating a runtime for.
+
+Next in your Dockerfile at the start add the docker image you want to base it off at the top like so:
+```bash
+FROM Dart:2.12 # Dart is used as an example.
+```
+This will download and require the image when you build your runtime and allow you to use the toolset of the language you are building a runtime for.
+
+Create a user and group for the runtime, this user will be used to both build and run the code:
+```bash
+RUN groupadd -g 2000 openruntimes \
+&& useradd -m -u 2001 -g openruntimes openruntimes
+```
+
+then create the folders you will use in your build step:
+```bash
+RUN mkdir -p /usr/local/src/
+RUN mkdir -p /usr/code
+RUN mkdir -p /usr/workspace
+RUN mkdir -p /usr/builtCode
+```
+
+Next copy your source code and set the working directory for the image like so:
+```
+WORKDIR /usr/local/src
+COPY . /usr/local/src
+```
+
+Next, you want to make sure you are adding execute permissions to any scripts you may run, the main ones are `build.sh` and `launch.sh`. You can run commands in Dockerfile's using the `RUN` prefix like so:
+```
+RUN chmod +x ./build.sh
+RUN chmod +x ./launch.sh
+```
+Note: Do not chmod a `launch.sh` file if you don't have one.
+
+If needed use the `RUN` commands to install any dependencies you require for the build stage.
+
+Next set the permissions for the user you created so your build and run step will have access to them:
+```
+RUN ["chown", "-R", "openruntimes:openruntimes", "/usr/local/src"]
+RUN ["chown", "-R", "openruntimes:openruntimes", "/usr/code"]
+RUN ["chown", "-R", "openruntimes:openruntimes", "/usr/workspace"]
+RUN ["chown", "-R", "openruntimes:openruntimes", "/usr/builtCode"]
+```
+
+Finally, you'll add a `CMD` command. For an interpreted language this should be:
+```
+CMD ["/usr/local/src/launch.sh"]
+```
+Since this will use your launch script when the runtime starts.
+
+For a compiled language this must be:
+```
+CMD ["tail", "-f", "/dev/null"]
+```
+so the build steps can be run.
+
+## 3. Building your Docker image and adding it to the list
+With your runtime successfully created you can now move on to building your docker image and adding it to the script files used for generating all of the image files.
+
+Open up the `/runtimes/buildLocalOnly.sh` script first and add your runtime to it. The following is an example with dart version 2.12
+```
+echo 'Dart 2.12...'
+docker build -t dart-runtime:2.12 ./runtimes/dart-2.12
+```
+Next, open up the `/runtimes/build.sh` script and also add your runtime to it. This one is slightly different as this is the one that will be used for cross-platform compiles and deploying it to Docker hub. The following is an example also with dart version 2.12:
```

Comment:
Please revisit this section, we're using `deploy.sh` to deploy 
and the environment variables come from `travis.yml`